modules/quest/src/main/java/unnamed/mmo/quest/objective/ParallelObjective.java

```diff
@@ -0,0 +1,79 @@
+package unnamed.mmo.quest.objective;
+
+import com.google.auto.service.AutoService;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.kyori.adventure.text.Component;
+import net.minestom.server.utils.NamespaceID;
+import net.minestom.server.utils.validate.Check;
+import org.jetbrains.annotations.NotNull;
+import unnamed.mmo.quest.QuestContext;
+import unnamed.mmo.quest.event.QuestObjectiveCompleteEvent;
+import unnamed.mmo.util.EventUtil;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static unnamed.mmo.util.ExtraCodecs.lazy;
+
+public record ParallelObjective(List<Objective> children) implements Objective {
+    public static final Codec<ParallelObjective> CODEC = RecordCodecBuilder.create(i -> i.group(
+            lazy(() -> Objective.CODEC).listOf().fieldOf("children").forGetter(ParallelObjective::children)
+    ).apply(i, ParallelObjective::new));
+
+    public ParallelObjective {
+        //todo not sure this should be an error? Probably should just complete instantly
+        Check.argCondition(children.isEmpty(), "children must not be empty");
+        Check.argCondition(children.size() > 32, "cannot have more than 32 children");
```

Comment:
Any particular reason 32 is the limit?

Reply:
I used the 32 bits of an integer to track whether each child is completed, so there is only space for 32. Could be changed to a long or a dynamic bit set in the future if needed. but 32 seemed like enough.